### ISSUE TITLE
--stdin-file-path=package.json doesn't work

### ISSUE THREAD

**Issue by haze** [biomejs/biome#1355]
When using biome on the CLI, using `--stdin-file-path=package.json` to format JSON through `stdin` does not work. Biome reports that `The content was not formatted because the formatter is currently disabled.` If I changed the file name to `a.json`, it works. This shouldn't matter as biome should only use this path to guess the extension.

### Environment information

```bash
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-ghostty"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.10.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
	"$schema": "https://biomejs.dev/schemas/1.4.1/schema.json",
	"organizeImports": {
		"enabled": true
	},
	"linter": {
		"enabled": true,
		"rules": {
			"recommended": true
		}
	}
}
```

### Playground link

N/A

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by haze** [comment]
This was introduced in https://github.com/biomejs/biome/pull/942 with the addition of `FILES_TO_NOT_PROCESS`.

**Comment by ematipico** [comment]
What's your expected result @haze ?

**Comment by haze** [comment]
I'd expect it to format my `package.json`, but going through the history, it seems that VSCode and other editors have a permissive JSON parser that lets them add comments. I don't know if biome supports this. My hack around this in the interim was just masking the file and using `file.js` as the path.

**Comment by ematipico** [comment]
The name of the file can be **anything**. Even _foo.json_. What's important is the file extension

**Comment by haze** [comment]
Correct, but if you run the command, the formatter won't work if the file name is any in the disallowed list. So... it *can* be anything, but it won't *work* with everything. If you run the command, it will spit out that it refused to format because of the file name. If you change the file name to `a.json` the warning goes away.

I only stumbled across this because the formatting plugin for emacs I'm using only gives me access to the filename in full. The first file I tried just happened to be `package.json`
